### PR TITLE
Convert data inputs into variable inputs for modules in Terraform

### PR DESCRIPTION
**Pull request by tanya-borisova**
# PR for issue https://github.com/microsoft/AzureTRE/issues/700

In several modules in Terraform, data inputs are used unnecessarily. Data inputs are intended for resources defined outside of Terraform, but in this case resources are defined within the same Terraform environment. This is also causing issues when Terraform thinks a resource has been updated when in reality it hasn't. 

## How is this addressed

- Remove `firewall/data.tf`, `gitea/terraform/data.tf`, `sonatype-nexus/terraform/data.tf`

**NOTE**: This is in draft for now because I'm resolving merge conflicts.

### CODE REVIEW

**Comment by marrobi** [comment]
We will be moving Gitea, Nexus and Firewall out as shared services to enable them to be managed as part of #23 . To be deployed separately after the core infrastructure. Fine to do this, but when we do that the data pieces will need to go back in.

Maybe I should create an issue and split them out into porter bundles to be installed via Make now to avoid you having to do this.

**Comment by tanya-borisova** [comment]
> Maybe I should create an issue and split them out into porter bundles to be installed via Make now to avoid you having to do this.

Sounds like a good idea to me

File: templates/shared_services/gitea/terraform/variables.tf [hunk 1]
```diff
@@ -47,8 +62,33 @@ variable "gitea_storage_limit" {
...
 }
+
+variable "firewall_name" {
+  type        = string
+  description = "Name of the firewall to connect to"
+}
+
+variable "firewall_resource_group_name" {
+  type        = string
+  description = "Name of the firewall to connect to"
```
Comment (jjgriff93) [inline templates/shared_services/gitea/terraform/variables.tf:93]: Name of the resource group containing the firewall resource

**Review by jjgriff93 — APPROVED** [review]

File: templates/shared_services/gitea/terraform/variables.tf (hunk 1 above)
Comment (tanya-borisova) [reply templates/shared_services/gitea/terraform/variables.tf:93]: Thanks for spotting this :)

**Pull request merged**